packages/jupyter-ai-magics/jupyter_ai_magics/models/persona.py

```diff
@@ -0,0 +1,42 @@
+import os
+from typing import ClassVar
+
+from langchain.pydantic_v1 import BaseModel
+
+JUPYTERNAUT_AVATAR_PATH = str(
+    os.path.join(os.path.dirname(__file__), "..", "static", "jupyternaut.svg")
+)
+JUPYTERNAUT_AVATAR_ROUTE = "api/ai/static/jupyternaut.svg"
+
+
+class Persona(BaseModel):
+    """
+    Model of an **agent persona**, a struct that includes the name & avatar
+    shown on agent replies in the chat UI.
+
+    Each persona is specific to a single provider, set on the `persona` field.
+    """
+
+    name: ClassVar[str] = ...
+    """
+    Name of the persona, e.g. "Jupyternaut". This is used to render the name
+    shown on agent replies in the chat UI.
+    """
+
+    avatar_route: ClassVar[str] = ...
+    """
+    The server route that should be used the avatar of this persona. This is
+    used to render the avatar shown on agent replies in the chat UI.
+    """
+
+    avatar_path: ClassVar[str] = ...
+    """
+    The path to the avatar SVG file on the server filesystem. The server should
+    serve the file at this path on the route specified by `avatar_route`.
+    """
```

Comment:
This is not used when defined by a custom provider, right?

Maybe it should be only defined in `JupyternautPersona`?

Reply:
The `avatar_path` field stores the path to the avatar icon that is requested by the frontend when loading a message. This field is used by any persona with an avatar that differs from the Jupyternaut icon. Therefore, custom providers need the `avatar_path` field to show a unique avatar for their provider.

Reply:
But my point is that this `avatar_path` field is not used anywhere except for hard-coded `JupyternautPersona.avatar_path`. Do you see what I mean?

It looks like custom providers exclusively can define `avatar_route` which is the only thing that gets passed to the frontend.

Reply:
@dlqqq 
Where is the `avatar_path` used other than hardcoding it in the route here? If I create a new provider with a new persona, and have a different path, where would that be useful? It seems like that for loading the avatar, only the route is being used.
```python
class AiExtension(ExtensionApp):
    name = "jupyter_ai"
	@@ -41,6 +46,14 @@ class AiExtension(ExtensionApp):
        (r"api/ai/providers?", ModelProviderHandler),
        (r"api/ai/providers/embeddings?", EmbeddingsModelProviderHandler),
        (r"api/ai/completion/inline/?", DefaultInlineCompletionHandler),
        # serve the default persona avatar at this path.
        # the `()` at the end of the URL denotes an empty regex capture group,
        # required by Tornado.
        (
            rf"{JUPYTERNAUT_AVATAR_ROUTE}()",
            StaticFileHandler,
            {"path": JUPYTERNAUT_AVATAR_PATH},
        ),
    ]
```

Reply:
@3coins @krassowski Thank you for the feedback. I agree it makes more sense for the avatar path to be specified in the extension serving the avatar rather than in the `Persona` definition itself. 

I pushed a commit that addresses this point of feedback by removing the `avatar_path` field. I've dropped the `PersonaDescription` model as well since it is identical to `Persona` after these changes.